internal/controller/postgrescluster/instance.go

```diff
@@ -1168,6 +1169,36 @@ func generateInstanceStatefulSetIntent(_ context.Context,
 		sts.Spec.Template.Spec.PriorityClassName = *spec.PriorityClassName
 	}
 
+	// if default pod scheduling is not explicitly disabled, add the default
+	// pod topology spread constraints
+	if cluster.Spec.DisableDefaultPodScheduling == nil ||
+		(cluster.Spec.DisableDefaultPodScheduling != nil &&
+			!*cluster.Spec.DisableDefaultPodScheduling) {
+		sts.Spec.Template.Spec.TopologySpreadConstraints = append(sts.Spec.Template.Spec.TopologySpreadConstraints,
+			corev1.TopologySpreadConstraint{
+				MaxSkew:           int32(1),
+				TopologyKey:       "kubernetes.io/hostname",
+				WhenUnsatisfiable: corev1.ScheduleAnyway,
+				LabelSelector: &metav1.LabelSelector{
+					MatchExpressions: []metav1.LabelSelectorRequirement{
+						{Key: naming.LabelCluster, Operator: "In", Values: []string{cluster.Name}},
+						{Key: naming.LabelData, Operator: "Exists"},
```

Comment:
🔧 This Exists should be `in (postgres, pgbackrest)`

Reply:
:+1: updated.